entrypoint.sh

```diff
@@ -113,9 +113,14 @@ create_uesr_data () {
     # ENCODED_USER_DATA=$(echo "$USER_DATA" | base64 | tr -d \\n)
     cat <<EOF > user_data.sh
 #!/bin/bash
+RUNNER_LABEL="self-hosted,linux,x86"
 if command -v yum &> /dev/null; then
+    # add rhel to the label
+    RUNNER_LABEL="rhel,${RUNNER_LABEL}"
     yum install -y curl jq libicu
 else
+    # add ubuntu to the label
+    RUNNER_LABEL="ubuntu,${RUNNER_LABEL}"
```

Comment:
ref https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#choosing-self-hosted-runners
> If you specify an array of strings or variables, your workflow will execute on any runner that matches all of the specified runs-on values. For example, here the job will only run on a self-hosted runner that has the labels linux, x64, and gpu:

IMO, when the test cases running in parallel, we will have three instances.
1. rhel,self-hosted,linux,x86
2. ubuntu, self-hosted,linux,x86 # 22.04
3. ubuntu, self-hosted,linux,x86 # 20.04

as our condition as belowing
https://github.com/sustainable-computing-io/aws_ec2_self_hosted_runner/blob/82cd501534ed0e7dcfbc9c2241ddbf858d200128/.github/workflows/ci_unit.yml#L61

how the label works?

Reply:
@SamYuan1990 yes, this is going to be interesting. Would you like to make a matrix that way? We don't have ubuntu  version tag (22.04 and 20.04 are the same in this sense), but rhel and ubuntu are now in the runner labels.

Reply:
IMO, I would like to use instance id to specific an instance for test.
@rootfs , could you please help test instance id can be a validate label or instance id as a hash value which too long for label?
if instance id is too long, we can use a label group as Operator system.

Reply:
it is not possible to know the instance id before hand so we cannot use it to set up tests. The instance type is known (e.g. t2.micro). See the attached image.